src/github.com/getlantern/fronted/direct.go

```diff
@@ -136,6 +137,8 @@ func (d *direct) Do(req *http.Request) (*http.Response, error) {
 // Dial persistently dials masquerades until one succeeds.
 func (d *direct) Dial(network, addr string) (net.Conn, error) {
 	gotFirst := false
+	candidateCh := <-candidateChCh
+	candidateChCh <- candidateCh
```

Comment:
Ahh you're saying because the `candidateChCh <- candidateCh` call will block if `Configure` sneaks in there and adds it's own channel, is that right? I guess we could also just throw it on a go routine and/or make candidateChCh a little bigger so it would never really happen in practice.

TBC, what value are you suggesting we make an atomic.Value @oxtoacart?